File: harmony-v0-arkts/entry/src/main/reference/android/android-v0-compose/app/src/main/java/com/example/v0clone/data/model/JobPreferenceModels.kt
package com.xlwl.AiMian.data.model

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class JobPreferenceDto(
    val positions: List<JobPreferencePositionDto> = emptyList()
) : Parcelable

@Parcelize
data class JobPreferencePositionDto(
    val id: String,
    val code: String,
    val name: String,
    val categoryId: String? = null,
    val categoryName: String? = null,
    val sortOrder: Int = 0
) : Parcelable

data class UpdateJobPreferencesRequest(
    val positionIds: List<String>
)

File: iOS-v0/Sources/Models/JobPreferenceModels.swift
import Foundation

struct JobPreference: Codable, Sendable {
  let positions: [JobPreferencePosition]
}

struct JobPreferencePosition: Codable, Identifiable, Hashable, Sendable {
  let id: String
  let code: String
  let name: String
  let categoryId: String?
  let categoryName: String?
  let sortOrder: Int
}

struct UpdateJobPreferencesRequest: Encodable, Sendable {
  let positionIds: [String]
}
